### PR TITLE
Add public example for HTCondor Pool

### PR DESCRIPTION
**Pull request by tpdownes**
This is blocked by PR #417 and #418. Once all 3 are merged I will follow up with

- PR to replace HTCondor integration test to use this example
- PR to remove functionality warnings from `htcondor-*` README

I will preemptively discuss a likely concern:

- OS Login is explicitly disabled because empirically HTCondor's internal notion of users fails to detect the username properly when using the extremely high UIDs generated by OS Login but works with UIDs ~1000 (or perhaps a problem with the `_` underscore characters); I interpret this an HTCondor bug which I will discuss with them

### Submission Checklist

* [X] Have you installed and run this change against pre-commit? (`pre-commit
  install`)
* [x] Are all tests passing? (`make tests`)
* [ ] Have you written unit tests to cover this change?
* [X] Is unit test coverage still above 80%?
* [X] Have you updated all applicable documentation?
* [X] Have you followed the guidelines in our Contributing document?

### CODE REVIEW

File: community/examples/htcondor-pool.yaml
Comment (nick-stroud) [inline community/examples/htcondor-pool.yaml:34]: nit: inline this: `outputs: [network_name]`

**Comment by tpdownes** [comment]
After rebasing when #418 was merged, I have force pushed. `make tests` succeeds locally. I unchecked the "make tests" box and re-checked it for good measure.

File: community/examples/htcondor-pool.yaml
Comment (nick-stroud) [inline community/examples/htcondor-pool.yaml:73]: nit: At one point we had talked about only being able to reference modules that had already been defined above (as a cheap way to avoid cycles). This would violate. Maybe move each startup script next to the module that is consuming them.

File: community/examples/htcondor-pool.yaml
Comment (nick-stroud) [inline community/examples/htcondor-pool.yaml:119]: No action: Part of me wants to see a htcondor access and htcondor cm modules. It would definitely be possible to make variable passing easier, cut down on some verbosity, and maybe even take the startup scripts inside of the module. With that said I think what you have here gets it down with much less overhead, so no objection.

**Review by nick-stroud — APPROVED** [review]

File: community/examples/htcondor-pool.yaml
Comment (tpdownes) [reply community/examples/htcondor-pool.yaml:119]: Upon original design we had not yet used GitHub URL based approach to use Toolkit modules. There is a need for enforcing the `central-manager` metadata existence and correctness. Maybe that could be addressed with `use`. I'll take a look / create a task.

File: community/examples/htcondor-pool.yaml
Comment (tpdownes) [reply community/examples/htcondor-pool.yaml:73]: Performed re-org due to increased clarity of connection.

File: community/examples/htcondor-pool.yaml
Comment (tpdownes) [reply community/examples/htcondor-pool.yaml:119]: Ack

File: community/examples/htcondor-pool.yaml
Comment (tpdownes) [reply community/examples/htcondor-pool.yaml]: Addressing this nit requires a re-approval @nick-stroud

**Review by nick-stroud — APPROVED** [review]

**Pull request merged**
